Azure/sonic-buildimage
# [DellEMC]: Platform modules Python3 compliance and other changes

**- Why I did it**

- Make DellEMC platform modules Python3 compliant.
- Change return type of PSU Platform APIs in DellEMC Z9264, S5232 and Thermal Platform APIs in S5232 to 'float'.
- Remove multiple copies of pcisysfs.py.
- PEP8 style changes for utility scripts.
- Build and install Python3 version of sonic_platform package.
- Fix minor Platform API issues.

**- How I did it**

- Used 2to3 and manual assessment for Python3 compliant changes.
- Made changes in thermal.py, psu.py to return 'float' value.
- Replaced multiple copies of pcisysfs.py with a single one in common directory.

**- How to verify it**

- Verified the values read/write by utility scripts. 
- Installed Python3 version of 'sonic_platform_base' and used a python script to load Chassis class and then call the APIs accordingly and verify the return type.

UT Logs: 
[S6100_py2to3_logs.txt](https://github.com/Azure/sonic-buildimage/files/5371244/S6100_py2to3_logs.txt) 
[S6000_py2to3_logs.txt](https://github.com/Azure/sonic-buildimage/files/5371252/S6000_py2to3_logs.txt)
[S5232_py2to3_logs.txt](https://github.com/Azure/sonic-buildimage/files/5371255/S5232_py2to3_logs.txt)
[Z9264_py2to3_logs.txt](https://github.com/Azure/sonic-buildimage/files/5371258/Z9264_py2to3_logs.txt)


**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**


[DellEMC]: Platform modules Python3 compliance, Platform API fixes and other changes.

**- A picture of a cute animal (not mandatory but encouraged)**
